content/javascript/concepts/objects/terms/groupBy/groupBy.md

```diff
@@ -0,0 +1,53 @@
+---
+Title: '.groupBy()'
+Description: 'Groups objects by a key.'
+Subjects:
+  - 'Web Development'
+Tags:
+  - 'Methods'
+  - 'Objects'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+The **`.groupBy()`** method groups items according to the value returned by a [callback function](https://codecademy.com/ressources/docs/javascript/callbacks). The value of the `.groupBy()` method is a null-prototype object: it doesn't inherit any object methods.
+
+## Syntax
+
+```pseudo
+Object.groupBy(items, callfunc);
+```
+
+The `items` are usually an array. They will be grouped according to the result of the `callfunc`.
+The `callfunc` is a callback function. This function has to return a string or a symbol.
+
+## Example
+
+In this example, we want to group users by their subscription year.
+
+```js
+const subs = [
+    { userName: 'Lisa', yearSub: 2022 },
+    { userName: 'Akim', yearSub: 2020 },
+    { userName: 'Lola', yearSub: 2020 },
+    ];
+
+const subsByYear = Object.groupBy(subs, user => {
+    return user.yearSub;
+    });
+```
+
```

Comment:
Add a note saying that the reader needs to add a library like Lodash to run this code.

Reply:
**Note:** To run this code, it is necessary to add a library like [Lodash](https://lodash.com/docs/4.17.15#groupBy).

Reply:
@cecilezan You need to add that in the content and not as a comment. You can visit Codecademy's [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) for that.